doc/crypto/api/library/library.rst

```diff
@@ -52,24 +83,156 @@ Library initialization
+.. macro:: PSA_CRYPTO_SUBSYSTEM_STORAGE
+    :definition: /* implementation-defined value */
+
+    .. summary::
+        Crypto subsystem identifier for access to keys in storage.
 
-        *   An implementation in which `psa_crypto_init()` initializes the random bit generator, and no operations that require the RNG can be performed until this is done. For example, random data, key, IV, or nonce generation; randomized signature or encryption; and algorithms that are implemented with blinding.
+    Initializing this subsystem and the `PSA_CRYPTO_SUBSYSTEM_KEYS` subsystem allows creating, accessing, and destroying persistent keys.
+
+    Persistent keys in secure elements also require `PSA_CRYPTO_SUBSYSTEM_SECURE_ELEMENTS`.
+
+.. macro:: PSA_CRYPTO_SUBSYSTEM_ACCELERATORS
+    :definition: /* implementation-defined value */
+
+    .. summary::
+        Crypto subsystem identifier for accelerator drivers.
+
+    Initializing this subsystem results in initialization of all registered accelerator drivers.
+
+    Initializing this subsystem allows cryptographic operations that are implemented via an accelerator driver.
+
+.. macro:: PSA_CRYPTO_SUBSYSTEM_SECURE_ELEMENTS
+    :definition: /* implementation-defined value */
+
+    .. summary::
+        Crypto subsystem identifier for secure element drivers.
+
+    Initializing this subsystem results in initialization of all registered secure element drivers.
+
+    Initializing this subsystem as well as `PSA_CRYPTO_SUBSYSTEM_KEYS` allows creating, accessing, and destroying keys in a secure element. That is, keys whose location is not `PSA_KEY_LOCATION_LOCAL_STORAGE`.
+
+.. macro:: PSA_CRYPTO_SUBSYSTEM_RANDOM
+    :definition: /* implementation-defined value */
+
+    .. summary::
+        Crypto subsystem identifier for the random generator.
+
+    Initializing this subsystem initializes all registered entropy drivers, and accesses the registered entropy sources.
+
+    Initializing this subsystem is necessary for `psa_generate_random()`, `psa_generate_key()`, and some operations using private or secret keys.
+
+    It is guaranteed that the following operations do not to require this subsystem:
+
+    *   Hash operations.
+    *   Signature verification operations.
+
+    Is it :scterm:`implementation defined` whether other operations require the initialization of this subsystem.
+
+.. macro:: PSA_CRYPTO_SUBSYSTEM_BUILTIN_KEYS
+    :definition: /* implementation-defined value */
+
+    .. summary::
+        Crypto subsystem identifier for access to built-in keys.
+
+    Initializing this subsystem as well as `PSA_CRYPTO_SUBSYSTEM_KEYS` allows access to built-in keys.
+
+.. macro:: PSA_CRYPTO_ALL_SUBSYSTEMS
+    :definition: /* implementation-defined value */
+
+    .. summary::
+        Crypto subsystem identifier for all available subsystems.
```

Comment:
I think this is effectively implied by the rule that using this identifier in a call to `psa_crypto_init_subsystem()` is equivalent to a call to `psa_crypto_init()`. But maybe we should spell that out, to avoid an implementation short cut that doesn't consider this kind of time evolution of the implementation?